components/brave_news/browser/feed_v2_builder.cc

```diff
@@ -1235,8 +1271,9 @@ mojom::FeedV2Ptr FeedV2Builder::GenerateAllFeed() {
 
   // Step 3: Generate a top news block
   // https://docs.google.com/document/d/1bSVHunwmcHwyQTpa3ab4KRbGbgNQ3ym_GHvONnrBypg/edit#heading=h.7z05nb4b269d
-  auto top_news_block =
-      GenerateChannelBlock(locale, publishers, kTopNewsChannel, articles);
+  // This block is a bit special - we take the top articles from the top few
+  // topics and display them in a cluster.
+  auto top_news_block = GenerateTopTopicsBlock(publishers, topics);
```

Comment:
this will only happen on the third block of the For You feed, right?

Reply:
and only once

Reply:
Yup, there'll be 1 Normal Block, one Ad, then this.

Reply:
what happens if topics are empty? Like in most locales

Reply:
Oh I see, we return an empty block. Could fall back to top news channel cluster if topics are empty instead?